### PR TITLE
Add `FalconCausalLM`

### PR DESCRIPTION
**Pull request by SamanehSaadat**
This PR adds `FalconCausalLM`. Here is a [Colab](https://colab.research.google.com/drive/15F_iCnsPU8WGf62aUhxunPqB-ggCSQ7-) for demo.

### CODE REVIEW

File: keras_nlp/src/models/falcon/falcon_causal_lm.py [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+        "falcon_refinedweb_1b_en",
+        preprocessor=None,
+    )
+    falcon_lm.fit(x=x, y=y, sample_weight=sw, batch_size=2)
+    ```
+
+    Custom backbone and vocabulary.
+    ```python
+    tokenizer = keras_nlp.models.FalconTokenizer(
+        proto="proto.spm",
```
Comment (mattdangerw) [inline keras_nlp/src/models/falcon/falcon_causal_lm.py:116]: I think this is not a sentence piece tokenizer, but gpt style bpe, so short merge and vocab here?

**Review by mattdangerw — APPROVED** [review]
Code looks great! Just a couple comments.

File: keras_nlp/src/models/falcon/falcon_causal_lm.py [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+
+    falcon_lm.compile(sampler=keras_nlp.samplers.BeamSampler(num_beams=2))
+    falcon_lm.generate("I want to say", max_length=30)
+    ```
+
+    Use `generate()` without preprocessing.
+    ```python
+    prompt = {
+        # Token ids for "<bos> Keras is".
+        "token_ids": np.array([[2, 214064, 603, 0, 0, 0, 0]] * 2),
```
[the remark this answers is not in the document]
Comment (SamanehSaadat) [reply keras_nlp/src/models/falcon/falcon_causal_lm.py]: Yeah, it was copied! Thanks for catching that!

File: keras_nlp/src/models/falcon/falcon_causal_lm.py (hunk 1 above)
Comment (SamanehSaadat) [reply keras_nlp/src/models/falcon/falcon_causal_lm.py]: Updated it to match the falcon tokenizer! Thanks for catching this!

**Review by mattdangerw — APPROVED** [review]
Thanks! Looks good to me!

**Pull request merged**